File: clients/javascript/game/player.js
const Cell = require('./cell.js');

module.exports = class Player {
  constructor(id, name, totalMass, active, cells) {
    this.id = id;
    this.name = name;
    this.totalMass = totalMass;
    this.active = active;
    this.cells = cells;
  }

  static parse(payload) {
    return new Player(
      payload.id,
      payload.name,
      payload.total_mass,
      payload.isActive,
      payload.cells.map(c => Cell.parse(c))
    );
  }

  get actions() {
    return this.cells.map(c => c.actions());
  }
};

File: clients/java/src/main/java/io/aigar/game/models/Player.java
package io.aigar.game.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class Player {
    private int id;
    private String name;
    @JsonProperty("total_mass")
    private int totalMass;
    private boolean isActive;
    private List<Cell> cells;

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getTotalMass() {
        return totalMass;
    }

    public boolean getIsActive() {
        return isActive;
    }

    public List<Cell> getCells() {
        return cells;
    }
}
